source/server/overload_manager_impl.cc

```diff
@@ -223,12 +278,14 @@ void OverloadManagerImpl::updateResourcePressure(const std::string& resource, do
                   const std::string& action = entry.second;
                   auto action_it = actions_.find(action);
                   ASSERT(action_it != actions_.end());
+                  const OverloadActionState old_state = action_it->second.getState();
                   if (action_it->second.updateResourcePressure(resource, pressure)) {
-                    const bool is_active = action_it->second.isActive();
-                    const auto state =
-                        is_active ? OverloadActionState::Active : OverloadActionState::Inactive;
-                    ENVOY_LOG(info, "Overload action {} became {}", action,
-                              is_active ? "active" : "inactive");
+                    const auto state = action_it->second.getState();
+
+                    if (old_state.isSaturated() != state.isSaturated()) {
+                      ENVOY_LOG(info, "Overload action {} became {}", action,
+                                (state.isSaturated() ? "saturated" : "scaling"));
+                    }
```

Comment:
is some logging appropriate for scaling state transitions?

Possibly LOG_EVERY_N_SEC when scaling, and/or LOG when entering and exiting scaling mode.

Reply:
Probably a good idea. Does Envoy have a LOG_EVERY_N_SEC macro?

Reply:
Doesn't seem like it.  At least not yet.

Reply:
This should also be at `debug` level I think.